app/models/settings/base.rb

```diff
@@ -0,0 +1,159 @@
+# This was taken from the rails-settings-cached gem and adapted to work with
+# more than one settings model by making the request cache class-specific. We
+# also removed features we don't need like cache scopes and readonly fields.
+#
+# See: https://github.com/huacnlee/rails-settings-cached
+module Settings
+  class Base < ActiveRecord::Base # rubocop:disable Rails/ApplicationRecord
+    self.abstract_class = true
+
+    class SettingNotFound < RuntimeError; end
+
+    class ProcetedKeyError < ArgumentError; end
```

Comment:
typo:

```suggestion
    class ProtectedKeyError < ArgumentError; end
```

though I see the original gem has this typo too :D https://github.com/huacnlee/rails-settings-cached/blob/7695c7f39b6616b7aa6c171425595e749da9955b/lib/rails-settings/base.rb#L4

Reply:
should these two classes be inside `Base`? Maybe `Settings::SettingNotFound` instead?

Reply:
Good call, I moved `ProtectedKeyError` to `Settings`. As for `SettingNotFound`, I removed it completely, it's never used, not even in the gem:

https://github.com/huacnlee/rails-settings-cached/search?q=settingnotfound